src/store.ts

```diff
@@ -41,8 +41,7 @@ const persistConfig = {
   key: "root",
   storage: AsyncStorage,
   stateReconciler: autoMergeLevel2,
-  // for some reason ` version: _.max(Object.keys(MIGRATIONS).map(parseInt)) ?? -1` always ends up as 0
-  version: parseInt(_.max(Object.keys(MIGRATIONS)) ?? "-1", 10),
+  version: _.max(Object.keys(MIGRATIONS).map(key => parseInt(key, 10))) ?? -1,
```

Comment:
Explanation: with the bare `.map(parseInt)` we were accidentally passing *two* parameters to `parseInt`: the key, and the index of the key. Since `parseInt` takes a second parameter (radix), we were effectively calling `parseInt("0", 0)` (which is `0`) and `parseInt("1", 1)` (which is `NaN`). This explicitly passes just the key with a radix of 10.

Note that what was committed here is actually taking the `max` of strings, not integers, so it'll rank `"2"` as higher than `"10"`. See this example code (can try it on https://playcode.io/lodash)

```js
import _ from 'lodash'

const MIGRATIONS = {
  0: "a",
  1: "b",
  2: "c",
  10: "d",
  11: "e",
}

console.log(parseInt(_.max(Object.keys(MIGRATIONS)) ?? "-1", 10));
console.log(_.max(Object.keys(MIGRATIONS).map(key => parseInt(key, 10))) ?? -1);
```

Reply:
Good catch.  Would have worked fine as long as we never have more than 10 schema versions ;-)